hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -149,7 +150,14 @@ public Address canonicalAddress() {
 		if (alias.isEmpty()) {
 			return id.asEvmAddress();
 		} else {
-			return Address.wrap(Bytes.wrap(alias.toByteArray()));
+			if (alias.size() == 20) {
+				return Address.wrap(Bytes.wrap(alias.toByteArray()));
+			} else if (alias.size() == 35) {
+				var addressBytes = EthTxSigs.recoverAddressFromPubKey(alias.toByteArray());
+				return addressBytes == null ? id.asEvmAddress() : Address.wrap(Bytes.wrap(addressBytes));
+			} else {
+				return id.asEvmAddress();
+			}
```

Comment:
It would be great to document these three cases -- why 20, why 35, and why else

Reply:
I'd suggest potentially adding adding 2 static variable `Address.ETHEREUM_ADDRESS_BYTE_SIZE` and `Address.ETHEREUM_PUBLIC_KEY_BYTE_SIZE` for easy use here and in `EntityIdUtils`.
Plus that was it's self documenting

Reply:
Done.